pkg/relayer/relayer.go

```diff
@@ -0,0 +1,49 @@
+package relayer
+
+import (
+	"context"
+
+	"cosmossdk.io/depinject"
+)
+
+type RelayerOption func(*Relayer)
+
+// Relayer is the main struct that encapsulates the relayer's responsibilities.
+// It starts and stops the RelayerProxy and provide the served relays observable to them miner.
+type Relayer struct {
```

Comment:
Thoughts on calling this struct (and the file) `RelayMiner`? @bryanchriswhite

Reply:
:joy: I would love that - haha. That's what we originally wanted to call it but IIRC there was some sort of decree that was made and "it shall be called 'the relayer'". 

To be clear on file naming, I would propose `relay_miner.go`.